api/v1/submissions/PKPSubmissionHandler.inc.php

```diff
@@ -901,4 +941,281 @@ public function deletePublication($slimRequest, $response, $args)
+        $author = Repo::author()->get($newId);
+
+        return $response->withJson(
+            Repo::author()->getSchemaMap()->map($author),
+            200
+        );
+    }
+
+    /**
+     * Delete one of this publication's contributors
+     *
+     * @param $slimRequest Request Slim request object
+     * @param $response Response object
+     * @param array $args arguments
+     *
+     * @return Response
+     */
+    public function deleteContributor($slimRequest, $response, $args)
+    {
+        $request = $this->getRequest();
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+
+        $publication = Repo::publication()->get((int) $args['publicationId']);
+        $author = Repo::author()->get((int) $args['contributorId']);
+
+        if (!$publication) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        if ($submission->getId() !== $publication->getData('submissionId')) {
+            return $response->withStatus(403)->withJsonError('api.publications.403.submissionsDidNotMatch');
+        }
+
+        if (!$author) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        if ($publication->getId() !== $author->getData('publicationId')) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        $output = Repo::author()->getSchemaMap()->map($author);
+
+        Repo::author()->delete($author);
+
+        return $response->withJson($output, 200);
+    }
+
+    /**
+     * Edit one of this publication's contributors
+     *
+     * @param $slimRequest Request Slim request object
+     * @param $response Response object
+     * @param array $args arguments
+     *
+     * @return Response
+     */
+    public function editContributor($slimRequest, $response, $args)
+    {
+        $request = $this->getRequest();
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+        $currentUser = $request->getUser();
+
+        $publication = Repo::publication()->get((int) $args['publicationId']);
+        $author = Repo::author()->get((int) $args['contributorId']);
+
+        if (!$publication) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        if (!$author) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        if ($submission->getId() !== $publication->getData('submissionId')) {
+            return $response->withStatus(403)->withJsonError('api.publications.403.submissionsDidNotMatch');
+        }
+
+        // Publications can not be edited when they are published
+        if ($publication->getData('status') === PKPSubmission::STATUS_PUBLISHED) {
+            return $response->withStatus(403)->withJsonError('api.publication.403.cantEditPublished');
+        }
```

Comment:
This check should be in place when adding or deleting a contributor as well as sorting.